pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -129,7 +145,10 @@
                             .setTimestamp(clock.millis())
                             .addAllProps(toPairs(schema.getProps()))
                             .build();
-                        return schemaStorage.put(schemaId, info.toByteArray(), context);
+                        
+                        return getMaxDeletedVersion(schemaId).thenCompose(deleteVersion ->
```

Comment:
I think we should return the maxDeletedVersion when we are doing compatibility check above. This would reduce the number of times we are reading schema info. So the logic would be something like this:

```
getSchema(schemaId).thenCompose(existingSchema -> {
    if (existingSchema == null) {
         return completedFuture(-1L, true)
    } else if (existingSchema.schema.isDeleted()) {
         return completedFuture(existingSchema.schema.version, true);
    } else {
         if (isTransitiveStrategy) {
              return ...;
         } else {
              // find the maxDeletedVersion in backward sequence.
              return ...;
         }
    }
}).thenCompose((maxDeletedVersion, isCompatible) -> {
})
```